cli/src/pcluster/validators/slurm_settings_validator.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+# with the License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+from enum import Enum
+
+from pcluster.validators.common import FailureLevel, Validator
+
+SLURM_SETTINGS_QUEUE_DENY_LIST = ["Nodes", "State", "PartitionName", "ResumeTimeout", "SuspendTime"]
+SLURM_SETTINGS_COMPUTE_RESOURCE_DENY_LIST = ["NodeName", "NodeHostName", "NodeAddr", "State", "Gres"]
+
+
+class SlurmCustomSettingLevel(str, Enum):
+    """
+    Slurm Custom Settings level.
+
+    This enum defines the scope where the custom settings are defined.
+    """
+
+    QUEUE = "Queue"
+    COMPUTE_RESOURCE = "ComputeResource"
+
+
+class SlurmCustomSettingsValidator(Validator):
+    """
+    Slurm Custom Settings validator.
+
+    Validate custom settings in Slurm ComputeResource and queue.
+    """
+
+    def _validate(self, custom_settings, deny_list, settings_level: SlurmCustomSettingLevel):
+        denied_settings = set()
+
+        for setting in deny_list:
+            if setting in custom_settings.keys():
+                denied_settings.add(setting)
+        if len(denied_settings) > 0:
+            settings = ",".join(sorted(denied_settings))
+            self._add_failure(
+                f"Setting or overriding the following custom Slurm settings at {settings_level} level is not allowed: "
```

Comment:
I would word it "Using the following custom Slurm settings...".

Reply:
I picked the working from the design doc, but I can update it if you think this version is better.
In this case I'd update also the doc.

Reply:
Fixed